pkg/controller/controller.go

```diff
@@ -548,6 +548,10 @@ func worker(queue workqueue.RateLimitingInterface, resourceType string, maxRetri
 					return false
 				}
 
+				for err != nil {
+					err = reconciler(key.(string))
+				}
+
```

Comment:
why is this required?

Reply:
Hi Gaurav, this loop is to wait until the reconcile error is resolved and the secret is created before proceeding. Seeing your comment, I realise that this loop will run even in the error scenario due to absence of machine class (not found or deleted).

As per my discussions with @hardikdr this is a temporary fix. A better and a more systematic fix with a retry limit is to be done in the near future.

Reply:
This is certainly a stopgap solution, we would eventually need a mechanism to exponentially-backoff for all the failing operations. 

Though I would suggest to not modify the worker-logic here, it could potentially block the worker-thread in the cases of errors.  

With this solution, we should simply trigger the reconciliation if the secret is NotFound, and that should be it. :)

Reply:
Okay, I will update it likewise.

Reply:
I have pushed my latest changes to reconcile it one more time if the secret referred by `machineClass` is `NotFound` instead of a wait loop.

Reply:
> Though I would suggest to not modify the worker-logic here, it could potentially block the worker-thread in the cases of errors.

I would second with Hardik on this. I don't think we should modify this part of the logic. Because this would affect reconcialtion of all workers. i.e - Machines, MachineSet, MachineDeployments, MachinesClasses. 

This logic could be moved to a higher level controller at MachineReconciler maybe? - https://github.com/gardener/machine-controller-manager/pull/495/commits/db647117c1930acddaf5d2e6ee2bbea0affce6ac.

Reply:
@ggaurav10 @hardikdr do provide your inputs as well.

Reply:
> With this solution, we should simply trigger the reconciliation if the secret is NotFound, and that should be it. :)

@hardikdr with this you mean get rid of this logic right? - https://github.com/gardener/machine-controller-manager/pull/495/files#diff-3c862eb54a8e0e161b534e0c67e5379eR545-R548. And just keep returning in other cases?

Reply:
Yes, that's true.